### ISSUE TITLE
Notification of Husqvarna: use Event-format-V2

### ISSUE THREAD

**Issue by MichelRabozee** [jeff-winn/homebridge-automower-platform#364]
Hi,

I just received a notification from Husqvarna related to their phasing out of Event-format-V1

> Dear User of Husqvarna Group Developer Portal. We want to inform you that we will launch our new IoT-backend in March 2025. Our current solution has grown old and earned its retirement. This will be a breaking change for those who are using our current solution, and you will need to change your apps according to the below information. The updated solution will be available from 20th December 2024. We will simultaneously run both solutions until 28th February 2025. After this the current version will be removed and only the updated version will be running. Please see the new event format here: Event-format-V2 

Is the current plugin version already using the new event format ?

Best regards,
Michel

**Environment:**

* **Plugin Version**: Event-format-V2
* **Homebridge Version**: v1.8.5
* **Node.js Version**: v20.17.0
* **Operating System**: macOS / hb-service

**Comment by jeff-winn** [comment]
It is not, and I got the same email. Looks like I’ve got some work to do. My mower is parked for the winter too, so I can’t test it either.

Thanks for reaching out, I’ll start digging in.

**Comment by MichelRabozee** [comment]
And mine is at the shop for yearly maintenance. Bad timing from Husqvarna to phase out in the winter 😟😟

**Comment by jeff-winn** [comment]
Finally started to dig into this, unfortunately due to Husqvarna using version 1 for both their V1 and V2 versions of the same API, the plugin is likely going to have to break, and be released after they roll out the new version. They're touching both the normal APIs as well as their streamed updates and the two likely won't be compatible with each other.

Best case scenario is update the code early, test it after they've released their change, and then release my update with the changes in place. Given my mower will be parked until spring, I'll be limited to a preview release but that's it.

If they'd done a v2 endpoint with all the v2 changes isolated, we wouldn't be forced to break but alas that isn't the case and there's nothing I can do about it.

**Comment by MichelRabozee** [comment]
Wow, that is not cool of Husqvarna :-( Having read (quickly) https://app.qa-developer-portal.dss.husqvarnagroup.net/apis/automower-connect-api?tab=websocket%20v2 I thought only the WebSocket stuff will be migrated (from WebSocket to WebSocket V2), and that in the transition period (from 20-dec-2024 till 1-mar-2025), both events (deprecated and v2, for instance "positions-event" and positions-event-v2) will be transmitted simultaneously, allowing the client applications to use the one they are able to process (so if the application understands already v2, it uses that one); for me the approach of not creating a new v2 endpoint was done to be able for applications to gradually implement the new events, without the need to do the whole move at once. Seems I was a bit too confident about how they implemented the move I guess :-(

**Comment by jeff-winn** [comment]
Yeah they didn’t only change their streaming events, the public API also changed to match some of their new event models and it’s just not worth trying to support both at once for only a few months when I can’t test it anyway.

The V1 of their API can return either V1 model or V2 model which isn’t compatible with V1. They didn’t version their API routes to v2 either so I’d just be guessing where the data is located for a very short runway.

I’m just working to jump straight to V2 at this point and not support both simultaneously.

**Comment by MichelRabozee** [comment]
OK, that seems to be the best approach. I am here for testing, but I do not know when I will get back my mowers (usually, the shop prepares them around end of January, but I need anyway to wait March for starting them. So, take your time and enjoy end of year quality time !!

**Comment by FlyingLemming** [comment]
Hello everyone!
Is there any news about this topic? I just got my 430X back from his winter residence and tested the plugin (2.3.0 preview 101) with the following findings:

1. Controlling the Automower with the switches works fine.
2. Upon (re-)start of the plugin, the mower status is retrieved correctly:

`[30.3.2025, 13:44:33] [YJ: Automower] Wert für 'Ankunftssensor' von 'Shaun' geändert: OPEN
[30.3.2025, 13:44:33] [YJ: Automower] Wert für 'Verlassenssensor' von 'Shaun' geändert: CLOSED
[30.3.2025, 13:44:33] [YJ: Automower] Wert für 'Bewegungssensor' von 'Shaun' geändert: NO_FAULT
[30.3.2025, 13:44:33] [YJ: Automower] Wert für 'Bewegungssensor' von 'Shaun' geändert: NOT_TAMPERED
[30.3.2025, 13:44:33] [YJ: Automower] Wert für 'Bewegungssensor' von 'Shaun' geändert: MOTION_DETECTED
[30.3.2025, 13:44:33] [YJ: Automower] Wert für 'Cutting Height' von 'Shaun' geändert: 5
[30.3.2025, 13:44:33] [YJ: Automower] Wert für 'Zeitplan' von 'Shaun' geändert: OFF
[30.3.2025, 13:44:33] [YJ: Automower] Wert für 'Pause' von 'Shaun' geändert: OFF
[30.3.2025, 13:44:33] [YJ: Automower] Mähererkennung abgeschlossen, 0 neue(n) Mäher gefunden.`

3. The streaming messages do not work yet. Neither sensor nor switch status is changed in accordance with the actual mower status. Instead, error messages referring to v2 are present in the log:

`[30.3.2025, 13:45:19] [YJ: Automower] Unbekanntes Ereignis empfangen: battery-event-v2
[30.3.2025, 13:45:19] [YJ: Automower] Unbekanntes Ereignis empfangen: position-event-v2
[30.3.2025, 13:45:19] [YJ: Automower] Unbekanntes Ereignis empfangen: mower-event-v2`

Does this match the current development status of the plugin? Is there anything I can do to assist?

**Comment by jeff-winn** [comment]
I’m just waiting for my own mower to come out of winter storage before I can work on it. I’m in Michigan in the US so normally around April the snow is done for sure. My guess is it’ll be about mid-April before I can really begin working on updating the plug-in.

Sorry for the inconvenience! For now, best to just use the onboard scheduler.

**Comment by jeff-winn** [comment]
Just wanted to post an update. The weather has finally turned warm enough overnight that the mower is out of storage and I’m repairing the boundary wire after the winter snow plows.

Got part of it fixed already, hopefully I’ll have it up and running soon and can begin making the needed updates.

Thanks for your patience!

**Comment by jeff-winn** [comment]
Finally had some time to look at this! The changes were absorbed pretty well, so hopefully I'll have this completed this weekend. I just have to update all the unit tests tomorrow and then run my usual set of tests with my mower as long as there's no storms.

Would you be interested in running the preview release as well once it's ready so it's not just me checking it?

**Comment by MichelRabozee** [comment]
Excellent ! Yes, of course, count me in for testing !

**Comment by jeff-winn** [comment]
Alright, the code has been merged, and the new version is published. It won't show an update is available though in the Homebridge UI, you'll have to manually update to the next version from the user interface.

Here's the instructions if you're unsure how to do it:
https://jeff-winn.github.io/homebridge-automower-platform/extras/prerelease-installation

The version you're looking for is: **v2.3.0-preview.105**

I did some testing prior to publishing the release, but it's late enough at night here that I'll need to complete testing tomorrow.

Let me know how it goes!

**Comment by MichelRabozee** [comment]
Hello, thank you very much.

I installed the new 105 preview. So far so good. I need to wait till Tuesday afternoon for full testing (the mower does primary zone on Monday, secondary zone on Tuesday).

I let you know the outcome.

**Comment by MichelRabozee** [comment]
Small thing so far: I have a warning because my NodeJS is version 22.xx.xx. Maybe the requirement should be updated to include NodeJS v22 (and maybe remove v18, because soon, v18 will be obsoleted if I am not mistaken).
```
[19/05/2025, 17:56:01] The plugin "homebridge-automower-platform" requires Node.js version of ^18.20.4 || ^20.16.0 which does not satisfy the current Node.js version of v22.15.1. You may need to upgrade your installation of Node.js - see https://homebridge.io/w/JTKEF
[19/05/2025, 17:56:01] The plugin "homebridge-automower-platform" requires Node.js version of ^18.20.4 || ^20.16.0 which does not satisfy the current Node.js version of v22.15.1. You may need to upgrade your installation of Node.js - see https://homebridge.io/w/JTKEF
```

**Comment by MichelRabozee** [comment]
Hello, I am please to report that everything is working as expected ! Excellent work, thank you.
